### PR TITLE
Fix of #87 - correctly handle function types in IdentifierNaming rule

### LINKED ISSUE #87: Error on argument with function type
On
```
fun ASTNode.findChildrenMatching(elementType: IElementType? = null, predicate: (ASTNode) -> Boolean): List<ASTNode> =
```

diktat fails with
```
[INFO]    [ktlint] C:\Users\pwx5333130\Documents\diKTat\diktat-rules\src\main\kotlin\org\cqfn\diktat\ruleset\utils\AstNodeUtils.kt:182:81: Internal Error (diktat-ruleset:identifier-naming). Please create a ticket at https://github.com/pinterest/ktlint/issues (if possible, provide the source code that triggered an error) (cannot be auto-corrected)
```

Offset corresponds to opening parentheses at `(ASTNode)`

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Fixed bug in logic
* Added tests

This pull request closes #87

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 1]
```diff
@@ -130,20 +133,24 @@ class IdentifierNaming : Rule("identifier-naming") {
...
+     *     * function type can have VALUE_PARAMETERs without name
      */
     private fun extractVariableIdentifiers(node: ASTNode): List<ASTNode> {
-        val destructingDeclaration = node.getFirstChildWithType(ElementType.DESTRUCTURING_DECLARATION)
+        val destructingDeclaration = node.getFirstChildWithType(DESTRUCTURING_DECLARATION)
         val result = if (destructingDeclaration != null) {
-            destructingDeclaration.getAllChildrenWithType(ElementType.DESTRUCTURING_DECLARATION_ENTRY)
+            destructingDeclaration.getAllChildrenWithType(DESTRUCTURING_DECLARATION_ENTRY)
                     .map { it.getIdentifierName()!! }
+        } else if (node.treeParent.elementType == VALUE_PARAMETER_LIST && node.treeParent.treeParent.elementType == FUNCTION_TYPE) {
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:146]: I am afraid that this check on `file level` can cause troubles. What if node (file level constant) will not have treeParent.treeParent?

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt]: Not sure, if such case can actually happen, but I have still added a check for this.

Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt]: so?

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt]: https://github.com/cqfn/diKTat/pull/117/commits/1d24f332024ed8f42f4233b1f3d7fb3be2c21ee0

**Pull request merged**